### ISSUE TITLE
Request: dual endstop feature for Y axis

### ISSUE THREAD

**Issue by Angel996** [MarlinFirmware/Marlin#6179]
I am using Marlin to control a diy laser CNC. This machine uses DUAL Y setup. I have enabled the DUAL Y stepper feature and it works. But it would be also nice to have a double endstop feature too, because it's especially important for laser cutters: mirror alignment will be ruined if Y carriages are not aligned.

So, I have just modified it to use a vacant endstop input, they are simply logic AND'ed to make sure Y homing ends only after both steppers are at home. This method makes the stepper which homes sooner, slide a bit. The modification is basically this:

temperature.cpp
```
#if HAS_Y_MIN
      if (READ(Y_MIN_PIN) && READ(Y2_MIN_PIN)) SBI(current_endstop_bits_local, Y_MIN);

```

endstops.cpp
```
#if HAS_Y_MIN
    SERIAL_PROTOCOLPGM(MSG_Y_MIN);
    SERIAL_PROTOCOLLN((((READ(Y_MIN_PIN) && READ(Y2_MIN_PIN))^Y_MIN_ENDSTOP_INVERTING) ? MSG_ENDSTOP_HIT : MSG_ENDSTOP_OPEN));

```
However, it would be nice to have it done the right way: each endstop should home their respective stepper separately. Once both are homed, homing should be flagged as ended. Would be a great improvement for everyone using Marlin to control a laser cutter.
